src/mame/drivers/r9751.cpp

```diff
@@ -127,6 +138,9 @@ class r9751_state : public driver_device
 	uint8_t m_term_data;
 	uint16_t m_serial_status;
 	uint16_t m_serial_status2;
+	std::queue<uint8_t> kbd_queue;
+	std::queue<uint16_t> serial_status_queue;
+	bool serial_input;
```

Comment:
Are you thinking about how you'll support save states?  Variable-length structures like an unbounded queue aren't conducive to this.  Do you have any clues as to what the actual queue depth is on the hardware and what it does on overrun?

Reply:
In this instance it's not necessary to save that state.  It's also a temporary solution to help understand the underlying command logic between the main cpu board and the smioc board.  Once enough of the structure is understood through testing in emulation it can be handled by the smioc's cpu.  This is the current focus right now, and much of it is embedded in custom undocumented chips and PALs.